### PR TITLE
first playwright test and new env changes

### PR DESCRIPTION
**Pull request by omehes**

### CODE REVIEW

**Comment by omehes** [comment]
@gregfitch to run tests, I use `./scripts/tests.local.sh`
The small playwright test might fail but at the moment the problem is `playwright._impl._api_types.Error: Executable doesn't exist at /home/seluser/.cache/ms-playwright/chromium-965416/chrome-linux/chrome`

**Comment by omehes** [comment]
@m1yag1 here are all the changes and my first playwright test for review

File: backend/app/tests/markers.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+# Markers need to be registered in conftest.py and added as a return value
+# in the get_custom_markers() function.
+
+nondestructive = mark.nondestructive
+parametrize = mark.parametrize
+xfail = mark.xfail
+
+slow = mark.slow
```
Comment (m1yag1) [inline backend/app/tests/markers.py:15]: Also, this markers.py file is a bit "legacy" and was for "syntactic sugar". This file is not necessary as we register markers in `conftest.py`.

File: backend/app/tests/ui/corgi/test_home_p.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import pytest
```
[the remark this answers is not in the document]
Comment (omehes) [reply backend/app/tests/ui/corgi/test_home_p.py:1]: this file does not exists anymore. weird that you can see it (I renamed it). corgi folder was created to distinguish away the new playwright tests, an easier conversion process for me. Once they are all converted, they will go where the selenium tests are (they will be then deleted).

File: backend/app/tests/ui/corgi/test_home_p.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+from pages.corgi.home import Home
+
+
+@pytest.mark.otto
```
[the remark this answers is not in the document]
Comment (omehes) [reply backend/app/tests/ui/corgi/test_home_p.py:6]: that mark des not exists either

File: backend/app/setup.cfg [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [tool:pytest]
-addopts=-vvvs --tb=long --showlocals --driver=Chrome
+addopts=-vvvs --tb=long --showlocals --driver=Chrome --html report.html
```
[the remark this answers is not in the document]
Comment (omehes) [reply backend/app/setup.cfg:2]: removed

File: backend/app/tests/markers.py (hunk 1 above)
Comment (omehes) [reply backend/app/tests/markers.py:15]: deleted

File: scripts/tests.local.sh [hunk 5]
```diff
@@ -35,4 +35,4 @@ docker-compose -f docker-stack.yml exec backend-tests wait-for-it -t 10 db:5432
 docker-compose -f docker-stack.yml exec db psql -h db -d postgres -U postgres -c "DROP DATABASE IF EXISTS tests"
 docker-compose -f docker-stack.yml exec db psql -h db -d postgres -U postgres -c "CREATE DATABASE tests ENCODING 'UTF8'"
 docker-compose -f docker-stack.yml exec backend-tests wait-for-it -t 10 backend:80
-docker-compose -f docker-stack.yml exec -T backend-tests pytest ./tests -vvv -m "ui or integration" --junitxml="${TEST_RESULTS}" --driver Chrome --base-url http://frontend --headless
+docker-compose -f docker-stack.yml exec -T backend-tests pytest ./tests -vvv -m ui --junitxml="${TEST_RESULTS}" --driver Chrome --base-url http://frontend --headless
```
[the remark this answers is not in the document]
Comment (omehes) [reply scripts/tests.local.sh:38]: I forgot to put it back on. Done

File: backend/app/tests/conftest.py [hunk 6]
```diff
@@ -1,5 +1,8 @@
 import os
 
+# Import fixtures
+pytest_plugins = ("fixtures.corgi_playwright",)
```
[the remark this answers is not in the document]
Comment (omehes) [reply backend/app/tests/conftest.py:4]: done

File: backend/app/fixtures/corgi_playwright.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+from playwright.sync_api import sync_playwright
```
[the remark this answers is not in the document]
Comment (omehes) [reply backend/app/fixtures/corgi_playwright.py:1]: oh, that was a remnant from my learning phase. Got used to it... Done

**Comment by omehes** [comment]
@m1yag1 made the changes and some more. this time all checks passed

File: backend/.gitignore [hunk 8]
```diff
@@ -1,18 +1,22 @@
-# Byte-compiled / optimized / DLL files
-__pycache__/
```
Comment (m1yag1) [inline backend/.gitignore:2]: These should not be removed from the .gitignore. we do not want to commit pycache files.

File: backend/.gitignore [hunk 9]
```diff
@@ -1,18 +1,26 @@
 # Byte-compiled / optimized / DLL files
 __pycache__/
 *.py[cod]
-*$py.class
 
-# C extensions
```
Comment (m1yag1) [inline backend/.gitignore:6]: I fail to see why any changes need to be made to the .gitignore based on the changes in your PR.

**Review by m1yag1 — APPROVED** [review]
Looks good but I don't think you need to change the .gitignore for any reason b/c of your changes.

**Review by m1yag1 — CHANGES_REQUESTED** [review]
Same as before.

**Comment by omehes** [comment]
@m1yag1 .gitignore is back to normal...

File: backend/app/tests/conftest.py [hunk 10]
```diff
@@ -1,5 +1,8 @@
 import os
 
+# Import fixtures
+pytest_plugins = ("fixtures.corgi",)
```
Comment (m1yag1) [inline backend/app/tests/conftest.py:4]: Also, the location of the fixtures file should be inside the tests folder.

**Review by m1yag1 — CHANGES_REQUESTED** [review]

File: backend/app/tests/conftest.py (hunk 10 above)
Comment (m1yag1) [reply backend/app/tests/conftest.py:4]: I did a quick change on my local. This is what I did:

Comment (omehes) [reply backend/app/tests/conftest.py]: This was again, copying from cnx-automation... :)

**Review by m1yag1 — APPROVED** [review]

**Comment by omehes** [comment]
@m1yag1 All checks passed. The 2 latest commits have the changes you requested

**Pull request merged**